src/utils/getSearchWith.ts

```diff
@@ -0,0 +1,21 @@
+import { Params } from '../types';
+
+export function getSearchWith(
+  params: Params,
+  search?: string | URLSearchParams,
+) {
+  const newParams = new URLSearchParams(search);
+
+  for (const [key, value] of Object.entries(params)) {
+    if (value === null) {
+      newParams.delete(key);
+    } else if (Array.isArray(value)) {
+      newParams.delete(key);
+      value.forEach(item => newParams.append(key, item.toString()));
+    } else {
+      newParams.set(key, value ? value.toString() : '');
```

Comment:
Here you're setting the value of the key to an empty string if the value is falsy. However, if the value is `0` or `false`, it might be better to keep these values, as they can be meaningful in certain contexts. Consider checking for `null` or `undefined` instead of falsy values.